### PR TITLE
Add uncrustify support for cpp files

### PR DESCRIPTION
**Pull request by VivianNK**
## Description

Add uncrustify support for cpp files; include .cpp file types in uncrustifyCheck, add indent requirement for cpp classes, and indentation requirement for externs.

Update the formatting using these policies in existing cpp files. Also apply to relevant header files (with cpp extern). 

For each item, place an "x" in between `[` and `]` if true. Example: `[x]`.
_(you can also check items in the GitHub UI)_

- [ ] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

<_Please describe the test(s) that were run to verify the changes._>

## Integration Instructions

<_Describe how these changes should be integrated. Use N/A if nothing is required._>

### CODE REVIEW

**Comment by makubacki** [comment]
Looks reasonable at a quick glance. I'll come back when the PR is published.

File: UnitTestFrameworkPkg/Test/GoogleTest/Sample/SampleGoogleTest/SampleGoogleTest.cpp [hunk 1]
```diff
@@ -51,37 +51,46 @@ TEST_F(GlobalBooleanVarTests, GlobalBooleanShouldBeChangeable) {
...
-    VOID  *SampleGlobalTestPointer = NULL;
-
-  protected:
-  void SetUp() override {
+public:
+  VOID *SampleGlobalTestPointer = NULL;
+
+protected:
+  void
+  SetUp (
```
Comment (VivianNK) [inline UnitTestFrameworkPkg/Test/GoogleTest/Sample/SampleGoogleTest/SampleGoogleTest.cpp:59]: I'm not a fan of this, personally, not sure why it does this when there are no parameters.

Comment (makubacki) [reply UnitTestFrameworkPkg/Test/GoogleTest/Sample/SampleGoogleTest/SampleGoogleTest.cpp:59]: It's specialized for edk2 style parameter formatting. Generally, we'd expect `VOID` to be here and it wouldn't be empty.

File: .pytool/Plugin/UncrustifyCheck/UncrustifyCheck.py
Comment (makubacki) [inline .pytool/Plugin/UncrustifyCheck/UncrustifyCheck.py:1]: The changes look okay to me. A lot of the changes are applying existing formatting rules to existing files. The code diff delta to edk2 will become a burden over time. So, I'd suggest we take the change to edk2 and then pull it into Project Mu as a cherry-pick.

To start that process, if you haven't already, can you make the Uncrustify config file and plugin change in edk2 and confirm the changes will not have any unexpected impact?

There's a command [here](https://github.com/tianocore/tianocore.github.io/wiki/EDK-II-Code-Formatting#manual-usage---generate-file-list-via-stdin) to run Uncrustify against all the files in the repo.

**Pull request closed without merging**